src/components/smart_objects/src/smart_schema.cc

```diff
@@ -59,8 +59,16 @@ ISchemaItemPtr CSmartSchema::getSchemaItem() {
 
 void CSmartSchema::applySchema(SmartObject& Object,
                                const bool remove_unknown_parameters,
-                               const utils::SemanticVersion& MessageVersion) {
+                               const utils::SemanticVersion& MessageVersion,
+                               rpc::ValidationReport* report__) {
   mSchemaItem->applySchema(Object, remove_unknown_parameters, MessageVersion);
+  if (remove_unknown_parameters) {
+    rpc::ValidationReport dummy_report("");
+    if (!report__) {
+      report__ = &dummy_report;
+    }
+    mSchemaItem->filterInvalidEnums(Object, MessageVersion, report__);
```

Comment:
Should the return value be checked here?

Reply:
I could leave a log here if that works, not much else we could do with it though in this case.

Reply:
Ah, sadly the logger is not available in the smart objects component, so there really isn't anything to do here.